File: SecondYear/Semester 4/Ujjwal/src/Assignment_3/Question_6.java
package Assignment_3;
//	Q6. Given an array of even and odd numbers, write a program to separate
//	even numbers from the odd numbers.	
public class Question_6 {
	public static void main(String[] args) {
		int[] arr = {1,2,1,2,1,0,0,0,1,0,1,0,1};
		int[] arr2 = compute(arr);
		for(int i=0 ; i<arr2.length ; i++) {
			System.out.println(arr2[i]);
		}
	}
	public static int[] compute(int[] arr) {
		int i=0;
		int j=0;
		int k=arr.length-1;
		while(j<=k) {
			if(arr[j]%2==0) {
				while(arr[i]%2==0)
					i++;
				swap(arr,i,j);
			}
			else {
				while(arr[k]==2)
					k--;
				swap(arr,j,k);
			}
			j++;
		}
		return arr;
	}
	public static int[] swap(int[] arr , int i , int j) {
		int temp = arr[i];
		arr[i] = arr[j];
		arr[j] = temp;
		return arr;
	}
}

File: ThirdYear/6thSemester/IDUSP/Assignment2/Question6.py
# 6. Write a program to find median of a given list of integers. Combine both odd and even number of
# terms.


def find_median(numbers):
    numbers.sort()
    length = len(numbers)
    if length % 2 == 0:
        middle_index1 = length // 2 - 1
        middle_index2 = length // 2
        median = (numbers[middle_index1] + numbers[middle_index2]) / 2
    else:
        middle_index = length // 2
        median = numbers[middle_index]
    return median

def main():
    numbers = [int(x) for x in input("Enter the list of integers separated by space: ").split()]
    median = find_median(numbers)
    print("Median:", median)

if __name__ == "__main__":
    main()
